servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -9984,16 +10017,35 @@ public void run() {
     }
 
     public void recordMetrics(ResourceContext ctx, int httpStatus) {
-        final String principalDomainName = getPrincipalDomain(ctx);
-        final String domainName = getRequestDomainName(ctx);
-        final Object timerMetric = getTimerMetric(ctx);
-        final String httpMethod = (ctx != null) ? ctx.getHttpMethod() : null;
-        final String apiName = (ctx != null) ? ctx.getApiName() : null;
-        final String timerName = (apiName != null) ? apiName + "_timing" : null;
-        metric.increment("zms_api", domainName, principalDomainName, httpMethod, httpStatus, apiName);
-        metric.stopTiming(timerMetric, domainName, principalDomainName, httpMethod, httpStatus, timerName);
+        try {
+            final String principalDomainName = getPrincipalDomain(ctx);
+            final String domainName = getRequestDomainName(ctx);
+            final Object timerMetric = getTimerMetric(ctx);
+            final String httpMethod = (ctx != null) ? ctx.getHttpMethod() : null;
+            final String apiName = (ctx != null) ? ctx.getApiName() : null;
+            final String timerName = (apiName != null) ? apiName + "_timing" : null;
+            metric.increment("zms_api", domainName, principalDomainName, httpMethod, httpStatus, apiName);
+            metric.stopTiming(timerMetric, domainName, principalDomainName, httpMethod, httpStatus, timerName);
+        } catch (Exception e) {
+            LOG.warn("Got exception during recordMetrics: {}", e.getMessage(), e);
+        }
     }
 
+    @Override
+    public void publishChangeMessage(ResourceContext ctx, int httpStatus) {
+        try {
+            if (httpStatus == 200) {
```

Comment:
with status only checking for 200, we'll miss notifications for created (201) and no content (204) which happens with deletes. Here we should check if httpStatus >= 200 && httpStatus <= 299